openfpga/src/fpga_bitstream/write_xml_fabric_bitstream.cpp

```diff
@@ -106,22 +107,60 @@ static int write_fabric_config_bit_to_xml_file(
     case CONFIG_MEM_STANDALONE:
     case CONFIG_MEM_SCAN_CHAIN:
       break;
-    case CONFIG_MEM_QL_MEMORY_BANK:
-    case CONFIG_MEM_MEMORY_BANK: {
-      /* Bit line address */
-      write_tab_to_file(fp, xml_hierarchy_depth + 1);
-      fp << "<bl address=\"";
-      for (const char& addr_bit : fabric_bitstream.bit_bl_address(fabric_bit)) {
-        fp << addr_bit;
+    case CONFIG_MEM_MEMORY_BANK:
+    case CONFIG_MEM_QL_MEMORY_BANK: {
+      if (fast_xml) {
+        // New way of printing XML
+        // This is fast (less than 100s) as compared to original 1300s seen in
+        // 100K LE FPFA
+        const FabricBitstreamMemoryBank* memory_bank =
+          fabric_bitstream.memory_bank_info();
+        /* Bit line address */
+        write_tab_to_file(fp, xml_hierarchy_depth + 1);
+        const fabric_bit_data& bit =
+          memory_bank->fabric_bit_datas[(size_t)(fabric_bit)];
+        const fabric_blwl_length& lengths =
+          memory_bank->blwl_lengths[bit.region];
+        if (bl_addr.size() == 0) {
+          VTR_ASSERT(wl_addr.size() == 0);
+          bl_addr.resize(lengths.bl);
+          wl_addr.resize(lengths.wl);
+          memset(&bl_addr[0], 'x', lengths.bl);
```

Comment:
Suggest to use std:string method rather than ``memset``. C-style data manipulation is not easy for junior engineers w.r.t. memory leak.

Reply:
Address via https://github.com/chungshien/OpenFPGA/commit/56d76741d5c55717279796d9220cbf3adc23c290